### ISSUE TITLE
CentOS 7 : Service no longer starts after halting and rebooting VM.

### ISSUE THREAD

**Issue by dnataraj** [chef-cookbooks/mysql#301]
I have this situation on Centos 7.0, Vagrant 1.7.2, ChefDK 0.3.6

I use the cookbook and provision MySQL 5.7 as follows 

```
mysql_service 'default' do
  version '5.7'
  bind_address '0.0.0.0'
  initial_root_password 'sdsdsdsddsdss'
  action [:create, :start]
end

mysql_client 'default' do
  action :create
end
```

And then I am able to SSH into this and do 

```
[vagrant@localhost ~]$ sudo service mysql-default status
Redirecting to /bin/systemctl status  mysql-default.service
mysql-default.service - mysql_service[mysql-default]
   Loaded: loaded (/usr/lib/systemd/system/mysql-default.service; enabled)
   Active: active (running) since Wed 2015-02-18 12:56:31 UTC; 1min 1s ago
  Process: 4305 ExecStartPost=/usr/libexec/mysql-default-wait-ready $MAINPID (code=exitadmroot2014ed, status=0/SUCCESS)
 Main PID: 4304 (mysqld_safe)
   CGroup: /system.slice/mysql-default.service
           ├─4304 /bin/sh /usr/bin/mysqld_safe --defaults-file=/etc/mysql-default/my.cnf --basedir=/usr
           └─4503 /usr/sbin/mysqld --defaults-file=/etc/mysql-default/my.cnf --basedir=/usr --datadir=/var/lib/mysql-default --plugin-dir=/usr/lib64/mysql/plugin --log...
```

But after I halt the VM (or package it with vagrant package, re-add the box and start it up), the service no longer starts 

```
[vagrant@localhost ~]$ sudo service mysql-default start
Redirecting to /bin/systemctl start  mysql-default.service
Job for mysql-default.service failed. See 'systemctl status mysql-default.service' and 'journalctl -xn' for details.

[vagrant@localhost ~]$ systemctl status mysql-default.service
mysql-default.service - mysql_service[mysql-default]
   Loaded: loaded (/usr/lib/systemd/system/mysql-default.service; enabled)
   Active: failed (Result: exit-code) since Wed 2015-02-18 12:59:53 UTC; 18s ago
  Process: 3250 ExecStartPost=/usr/libexec/mysql-default-wait-ready $MAINPID (code=exited, status=1/FAILURE)
  Process: 3249 ExecStart=/usr/bin/mysqld_safe --defaults-file=/etc/mysql-default/my.cnf --basedir=/usr (code=exited, status=0/SUCCESS)
 Main PID: 3249 (code=exited, status=0/SUCCESS)

[vagrant@localhost ~]$ sudo journalctl -xn
-- Logs begin at Wed 2015-02-18 12:59:11 UTC, end at Wed 2015-02-18 13:00:57 UTC. --
Feb 18 12:59:50 localhost.localdomain mysqld_safe[3249]: 150218 12:59:50 mysqld_safe Logging to '/var/log/mysql-default/error.log'.
Feb 18 12:59:50 localhost.localdomain mysqld_safe[3249]: mkdir: cannot create directory ‘/var/run/mysql-default’: Permission denied
Feb 18 12:59:50 localhost.localdomain mysqld_safe[3249]: chown: cannot access ‘/var/run/mysql-default’: No such file or directory
Feb 18 12:59:50 localhost.localdomain mysqld_safe[3249]: chmod: cannot access ‘/var/run/mysql-default’: No such file or directory
Feb 18 12:59:50 localhost.localdomain mysqld_safe[3249]: 150218 12:59:50 mysqld_safe Starting mysqld daemon with databases from /var/lib/mysql-default
Feb 18 12:59:52 localhost.localdomain mysqld_safe[3249]: 150218 12:59:52 mysqld_safe mysqld from pid file /var/run/mysql-default/mysqld.pid ended
Feb 18 12:59:53 localhost.localdomain systemd[1]: mysql-default.service: control process exited, code=exited status=1
Feb 18 12:59:53 localhost.localdomain systemd[1]: Failed to start mysql_service[mysql-default].
-- Subject: Unit mysql-default.service has failed
-- Defined-By: systemd
-- Support: http://lists.freedesktop.org/mailman/listinfo/systemd-devel
-- 
-- Unit mysql-default.service has failed.
-- 
-- The result is failed.
Feb 18 12:59:53 localhost.localdomain systemd[1]: Unit mysql-default.service entered failed state.
Feb 18 13:00:57 localhost.localdomain sudo[3502]: vagrant : TTY=pts/0 ; PWD=/home/vagrant ; USER=root ; COMMAND=/bin/journalctl -xn
```

I have noticed that /etc/default/mysql-default/my.cnf has /var/run paths, but /var/lib paths are expected. Correcting this seems to fix the problem, but 
```
sudo service mysql-default start
```
does not return anymore (but service is running etc.)

TL;DR After I provision MySQL, I'm able to SSH and start/stop the MySQL service - but if I start/stop the VM (or package it and use vagrant up again), the service fails to start.

**Comment by dnataraj** [comment]
I have found a solution to my issues 
```
mysql_service 'default' do
  ...
  socket '/var/lib/mysql/mysqld.sock'
  ...
end
```
Seems to set the path to the socket correctly in /usr/libexec/mysql-default-wait-ready
and I also provided by own my.cnf 
```
cookbook_file '/etc/mysql-default/my.cnf' do
  source 'my.cnf'
  owner 'mysql'
  group 'mysql'
  mode '0600'
end
```
which had correct paths. It looks like the first time the VM boots, the socket and pid is set up in /var/run/mysql-default as root but once to stop and try restarting, the mysql user has no rights to set up things in /var/run.

**Comment by someara** [comment]
Derp. tmpfs on /run type tmpfs (rw,nosuid,nodev,mode=755)

**Comment by someara** [comment]
Fixed in master. Added tmpfiles.d templates so the run directories are created without Chef on reboot.
Will ship in next release.
